marc.py

```diff
@@ -592,52 +592,91 @@ def create_record(cls, work, annotator, force_create=False, integration=None):
 
         return record
 
-    def records(self, lane, annotator, start_time=None, force_refresh=False, mirror=None):
-        output = StringIO()
+    def records(self, lane, annotator, start_time=None, force_refresh=False, mirror=None, query_batch_size=500, upload_batch_size=7500):
+        """
+        Create and export a MARC file for the books in a lane.
+
+        :param lane: The Lane to export books from.
+        :param annotator: The Annotator to use when creating MARC records.
+        :param start_time: Only include records that were created or modified after this time.
+        :param force_refresh: Create new records even when cached records are available.
+        :param mirror: Optional mirror to use instead of loading one from configuration.
+        :param query_batch_size: Number of works to retrieve with a single database query.
+        :param upload_batch_size: Number of records to mirror at a time. This is different
+          from query_batch_size because S3 enforces a minimum size of 5MB for all parts
+          of a multipart upload except the last, but 5MB of records would be too many
+          works for a single query.
+        """
+
+        # We mirror the content, if it's not empty. If it's empty, we create a CachedMARCFile
+        # and Representation, but don't actually mirror it.
+        if not mirror:
+            storage_protocol = self.integration.setting(self.STORAGE_PROTOCOL).value
+            if storage_protocol == ExternalIntegration.S3:
+                mirror_integration = ExternalIntegration.lookup(
+                    self._db, storage_protocol, ExternalIntegration.STORAGE_GOAL)
+                if mirror_integration:
+                    mirror = mirror or S3Uploader(mirror_integration)
```

Comment:
This seems very similar to `MirrorUploader.sitewide`. Can you use that instead?

Reply:
I think we talked about this in my first MARC PR too. I had originally thought of having an option to store MARC files on disk so it wasn't necessary to set up S3, but I'm not sure if anyone will need that. I didn't like the assumption that there would only ever be one storage option configured though.

Reply:
When we support more than one storage option, this code will probably need to be changed anyway. This code implies that there can be multiple storage options, but only one per protocol.

Not a big deal -- we can revisit this when we add other storage options. My only concern is that when that happens I will probably go through the code looking for references to sitewide() and I might miss this one.